net: wifi: Fix user request disconnection

kapbh opened the pull request:
When user tries to disconnect the shell, it's always printing "Disconnection request failed".

krish2718 commented on include/zephyr/net/wifi_mgmt.h:
```diff
@@ -389,10 +389,10 @@ enum wifi_conn_status {
 	/** Disconnected due to user request */
 	WIFI_REASON_DISCONN_USER_REQUEST,
+	/** Unspecified reason */
+	WIFI_REASON_DISCONN_UNSPECIFIED = WIFI_STATUS_DISCONN_FIRST_STATUS,
```
instead of this mishmash, should we instead add a check in the shell to deal with this?

kapbh replied:
Actually this change will help to use WIFI_REASON_DISCONN_USER_REQUEST of Enum wifi_disconn_reason. Currently it is unused. By changing to 0 this will place disconnection request successfully.

jukkar replied:
This looks very weird now, the `WIFI_REASON_DISCONN_USER_REQUEST` gets value 0, and the unspecified reason gets a value after the last connection status value. This looks very confusing, why would we need to do anything like this in the first place? Why not check this thing in shell as suggested by @krish2718

kapbh replied:
When user tries to disconnect using wifi disconnect command the disconnection result is getting WIFI_REASON_DISCONN_UNSPECIFIED only. The idea is to use WIFI_REASON_DISCONN_USER_REQUEST for user to disconnect. For every non 0 value, disconnection result will be show as failed. For that making WIFI_REASON_DISCONN_USER_REQUEST as a 0.

Still if don't want to change that, I can make change in wifi shell condition, and check if it's WIFI_STATUS_CONN_LAST_STATUS than disconnect request will be successful.

jukkar replied:
> Still if don't want to change that, I can make change in wifi shell condition, and check if it's WIFI_STATUS_CONN_LAST_STATUS than disconnect request will be successful.

Yes, I think this is a better solution as it is causing less confusion to anyone reading the code.

krish2718 commented on subsys/net/l2/wifi/wifi_shell.c:
```diff
@@ -242,7 +242,7 @@ static void handle_wifi_disconnect_result(struct net_mgmt_event_callback *cb)
 
 	if (context.disconnecting) {
-		if (status->status) {
+		if (status->status != WIFI_STATUS_CONN_LAST_STATUS) {
```
```suggestion
		if (status->status != WIFI_REASON_DISCONN_UNSPECIFIED) {
```

Even this isn't quite right, we need to have a separate (status, reason), and if status is failure then reason comes in to play.

kapbh committed:
net: wifi: Add disconnection success code

When disconnect request is successful, the status
was taking as WIFI_REASON_DISCONN_UNSPECIFIED only.
Adding WIFI_REASON_DISCONN_SUCCESS which can be help
to determine status of disconnect request. If this status
is failed then reason can be useful.

Signed-off-by: Kapil Bhatt <kapil.bhatt@nordicsemi.no>
